### PR TITLE
ITOS-147: Add Institution Frequenting By User report

### PR DESCRIPTION
**Pull request by kcissewski**

### CODE REVIEW

File: reports/institutionFrequentingByUser.jrxml [hunk 1]
```diff
@@ -0,0 +1,295 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jasperReport xmlns="http://jasperreports.sourceforge.net/jasperreports" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://jasperreports.sourceforge.net/jasperreports http://jasperreports.sourceforge.net/xsd/jasperreport.xsd" name="numberPatientsBySex" pageWidth="612" pageHeight="792" whenNoDataType="AllSectionsNoDetail" columnWidth="572" leftMargin="20" rightMargin="20" topMargin="20" bottomMargin="20" uuid="df48aa2c-3795-40ea-8bb4-7e8988796c80">
```
Comment (areklalo) [inline reports/institutionFrequentingByUser.jrxml]: Please change the name of the report.

File: reports/institutionFrequentingByUser.jrxml [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+		<box>
+			<pen lineWidth="0.5" lineColor="#000000"/>
+		</box>
+	</style>
+	<subDataset name="dataset1" uuid="c19a8940-5445-4aac-95be-401c7d870329">
+		<parameter name="periods_iso" class="java.lang.String"/>
+		<parameter name="organisationunits_uid" class="java.lang.String"/>
+		<queryString>
+			<![CDATA[SELECT usr.username as User,
+	COUNT(DISTINCT CASE WHEN (p.vih_status = 1 AND DATE_FORMAT(p.date_created, "%Y%m") = $P{periods_iso})
```
Comment (areklalo) [inline reports/institutionFrequentingByUser.jrxml]: I believe there is missing one of the columns (which contains the visit date).

File: reports/institutionFrequentingByUser.jrxml [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+		THEN p.patient_id  else null END) AS 'Nouveau Patient',
+	COUNT(DISTINCT CASE WHEN (p.vih_status = 1 AND DATE_FORMAT(enc.encounter_datetime, "%Y%m") = $P{periods_iso})
+		THEN p.patient_id else null END) AS 'Patient VIH',
+	COUNT(DISTINCT CASE WHEN (p.vih_status = 0 AND DATE_FORMAT(enc.encounter_datetime, "%Y%m") = $P{periods_iso})
+		THEN p.patient_id else null END) AS 'Patient Non VIH',
+	COUNT(DISTINCT CASE WHEN ((p.vih_status = 1 OR p.vih_status = 0 ) AND DATE_FORMAT(enc.encounter_datetime, "%Y%m") = $P{periods_iso})
+		THEN p.patient_id else null END) AS 'Total Patient'
+	FROM openmrs.users usr INNER JOIN isanteplus.patient p ON usr.user_id = p.creator LEFT OUTER JOIN openmrs.encounter enc
+		ON p.patient_id = enc.patient_id, isanteplus.org_code_uid org
+	WHERE ((DATE_FORMAT(enc.encounter_datetime, "%Y%m") = $P{periods_iso})
```
Comment (ubap) [inline reports/institutionFrequentingByUser.jrxml]: I belive this should be
`	WHERE ((DATE_FORMAT(enc.encounter_datetime, "%Y%m") = $P{periods_iso}
		OR (DATE_FORMAT(p.date_created, "%Y%m") = $P{periods_iso})))
		AND p.organisation_uid = org.uid
		AND FIND_IN_SET($P{organisationunits_uid}, org.path)
`

https://stackoverflow.com/questions/1241142/sql-logic-operator-precedence-and-and-or

File: reports/institutionFrequentingByUser.jrxml (hunk 2 above)
Comment (ubap) [reply reports/institutionFrequentingByUser.jrxml]: As discussed offline, this will be addressed in ITOS-211

**Comment by ubap** [comment]
Fixed report name.

File: reports/institutionFrequentingByUser.jrxml (hunk 2 above)
Comment (areklalo) [reply reports/institutionFrequentingByUser.jrxml]: Ok, thank you

**Pull request merged**